src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs

```diff
@@ -253,53 +253,7 @@ private void OnCheckSocketConnection(object? state)
-
-                        // Poll socket to check if it can be read from. Unfortunatly this requires reading pending data.
-                        // The server might send data, e.g. HTTP/2 SETTINGS frame, so we need to read and cache it.
-                        //
-                        // Available data needs to be read now because the only way to determine whether the connection is closed is to
-                        // get the results of polling after available data is received.
-                        bool hasReadData;
-                        do
-                        {
-                            closeSocket = IsSocketInBadState(socket, socketAddress);
-                            var available = socket.Available;
-                            if (available > 0)
-                            {
-                                hasReadData = true;
-                                var serverDataAvailable = CalculateInitialSocketDataLength(_initialSocketData) + available;
-                                if (serverDataAvailable > MaximumInitialSocketDataSize)
-                                {
-                                    // Data sent to the client before a connection is started shouldn't be large.
-                                    // Put a maximum limit on the buffer size to prevent an unexpected scenario from consuming too much memory.
-                                    throw new InvalidOperationException($"The server sent {serverDataAvailable} bytes to the client before a connection was established. Maximum allowed data exceeded.");
-                                }
-
-                                SocketConnectivitySubchannelTransportLog.SocketReceivingAvailable(_logger, _subchannel.Id, socketAddress, available);
-
-                                // Data is already available so this won't block.
-                                var buffer = new byte[available];
-                                var readCount = socket.Receive(buffer);
-
-                                _initialSocketData ??= new List<ReadOnlyMemory<byte>>();
-                                _initialSocketData.Add(buffer.AsMemory(0, readCount));
-                            }
-                            else
-                            {
-                                hasReadData = false;
-                            }
-                        }
-                        while (hasReadData);
-                    }
-                    catch (Exception ex)
-                    {
-                        closeSocket = true;
-                        checkException = ex;
-                        SocketConnectivitySubchannelTransportLog.ErrorCheckingSocket(_logger, _subchannel.Id, socketAddress, ex);
-                    }
+                    closeSocket = CheckSocket(socket, socketAddress, ref _initialSocketData, out checkException);
```

Comment:
It's not a change, but I'm a little surprised this call site has less logging than the other one.

Reply:
I added a lot of logging in to the other call site while trying to figure out this bug 😄